etcdnaming/etcdnaming.go

```diff
@@ -0,0 +1,144 @@
+package etcdnaming
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
+	"sync"
+
+	etcdcl "github.com/coreos/etcd/client"
+)
+
+type namePair struct {
+	key, value string
+}
+
+// recvBuffer is an unbounded channel of item.
+type recvBuffer struct {
+	c        chan *namePair
+	mu       sync.Mutex
+	stopping bool
+	backlog  []*namePair
+}
+
+func newRecvBuffer() *recvBuffer {
+	b := &recvBuffer{
+		c: make(chan *namePair, 1),
+	}
+	return b
+}
+
+func (b *recvBuffer) put(r *namePair) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	b.backlog = append(b.backlog, r)
+	if !b.stopping {
```

Comment:
this check should be above the backlog op in line 33. Probably something like:

if b.stopping {
  return
}
b.backlog = append(...)
...